bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieStateManager.java

```diff
@@ -149,6 +160,30 @@ public Number getSample() {
             }
         };
         statsLogger.registerGauge(SERVER_STATUS, serverStatusGauge);
+        this.serverSanityGauge = new Gauge<Number>() {
+            @Override
+            public Number getDefaultValue() {
+                return -1;
+            }
+
+            @Override
+            public Number getSample() {
+                return sanityPassed.get();
+            }
+        };
+        statsLogger.registerGauge(SERVER_SANITY, serverSanityGauge);
+        stateService.scheduleAtFixedRate(() -> {
+            if (isReadOnly()) {
+                sanityPassed.set(1);
+                return;
+            }
+            SanityTestCommand.handleAsync(conf, new SanityTestCommand.SanityFlags()).thenAccept(__ -> {
```

Comment:
It looks like the config `sanityCheckMetricsEnabled` doesn't work. The scheduled task only works when `sanityCheckMetricsEnabled` is `true`.